### PR TITLE
Add KServe support

### PR DESCRIPTION
**Pull request by ruivieira**
This PR adds support to KServe inference services (IS) by setting a TrustyAI service as an `InferenceLogger` for KServe IS.
If IS is part of a ModelMesh deployment, the behaviour is the same as previously.

KServe and ModelMesh IS are differentiated by checking the IS's `serving.kserve.io/deploymentMode: ModelMesh` annotation.

**Other changes**

- Dependencies updated to better align with the ODH operator
- Updated KServe API from `v1alpha1` to `v1beta1`

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -107,6 +107,27 @@ rules:
...
+  resources:
+  - inferenceservices
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
```
Comment (danielezonca) [inline config/rbac/role.yaml:121]: Are really all of them necessary? For example I don't think `create` or `delete` are necessary

**Review by danielezonca — APPROVED** [review]

File: config/rbac/role.yaml (hunk 1 above)
Comment (ruivieira) [reply config/rbac/role.yaml]: Removed

**Pull request merged**